platform_umbrella/apps/common_core/lib/common_core/resources/postgres/cloudnative_pg_clusters.ex

```diff
@@ -198,15 +198,11 @@ defmodule CommonCore.Resources.CloudnativePGClusters do
   defp secret_data(state, cluster, user) do
     hostname = PostgresState.read_write_hostname(state, cluster)
 
-    password_version =
-      cluster.password_versions
-      |> Enum.sort_by(& &1.version, :desc)
-      |> Enum.find(cluster.password_versions, &(&1.username == user.username))
+    password = PostgresState.password_for_user(state, cluster, user)
```

Comment:
I copied this logic into a shared space so I could reuse it. It actually had a subtle bug.

The last clause was really `Enum.find(sorted_versions, cluster.password_versions, search)`. So the full, unsorted list was the default if nothing was found and the nil check below would never fail.